### PR TITLE
adds more apis and compatibility tests

### PR DESCRIPTION
**Pull request by Abhishekvrshny**

### CODE REVIEW

File: internal/project/validation.go [hunk 1]
```diff
@@ -21,8 +21,8 @@ func init() {
 }
 
 // GetValidatedModel validates an incoming proto request and returns a project model
-func GetValidatedModel(ctx context.Context, req *metrov1.Project) (*Model, error) {
-	if req.Name == "" {
+func GetValidatedModel(ctx context.Context, req *metrov1.Project, allowEmptyName bool) (*Model, error) {
```
Comment (sunnyagg) [inline internal/project/validation.go:24]: this seems hackish, why do we need project with empty name

Comment (Abhishekvrshny) [reply internal/project/validation.go:24]: delete call takes projectID in the URL path, where project name can be empty. But when creating a project, name cannot be empty. But in both cases, I am preparing a model to create or delete the entity and hence this way

Comment (sunnyagg) [reply internal/project/validation.go:24]: should be have different validators for different operations than combining into one method

File: internal/subscription/core.go [hunk 2]
```diff
@@ -72,3 +75,20 @@ func (c *Core) Exists(ctx context.Context, key string) (bool, error) {
 	}
 	return ok, nil
 }
+
+// DeleteSubscription deletes a subscription
+func (c *Core) DeleteSubscription(ctx context.Context, m *Model) error {
```
Comment (sunnyagg) [inline internal/subscription/core.go:80]: should we also validate if the subscription belongs to a project as well

Comment (Abhishekvrshny) [reply internal/subscription/core.go:80]: This is being checked here. https://github.com/razorpay/metro/pull/44/files#diff-62bfaba279adf117168ffb1d9690cc9c5d7227ad67cd181eb9934b4330bdf779R87 
key() returns the full consul path registry/projects/<project>/subscriptions/<subscription>

File: internal/project/validation.go (hunk 1 above)
Comment (Abhishekvrshny) [reply internal/project/validation.go]: done

**Comment by Abhishekvrshny** [comment]
@sunnyagg addressed the review comments

**Review by sunnyagg — APPROVED** [review]

**Pull request merged**